packages/syft/src/syft/grid/client/dataset.py

```diff
@@ -0,0 +1,60 @@
+# stdlib
+from pathlib import Path
+import platform
+import tarfile
+import tempfile
+from typing import Optional
+from typing import Union
+
+# third party
+import numpy as np
+import torch as th
+
+
+class Dataset:
+    def __init__(
+        self,
+        data: Union[th.Tensor, list, np.ndarray],
+        description: Optional[str] = None,
+        tags: Optional[list] = None,
+    ) -> None:
+        """WARNING: this should only ever be callable on a CLIENT
+        by a DATA OWNER. This class should NEVER BE DESERIALIZED ONTO A WORKER.
+        It INTERACTS WITH THE FILE SYSTEM!
+        This is a convenience class just to make uploading datasets to PyGrid
+        easier to do from a notebook until we have the ability to upload directly.
+        """
+        if isinstance(data, th.Tensor):
+            data = data.numpy()
+        elif isinstance(data, list):
+            data = np.array(data)
+        self.data = data
+        self.desc = description
+        self.tags = tags
+
+    def tozip(self) -> Path:
+        tempdir = Path(
+            "/tmp" if platform.system() == "Darwin" else tempfile.gettempdir()
+        )
+        tempdir = str(tempdir.absolute())
+        np.savetxt(tempdir + "/data_01.csv", self.data, delimiter=",")
+        f = open(tempdir + "/manifest", "w")
+        f.write(self.desc)
+        f.close()
+        f = open(tempdir + "/description", "w")
+        f.write(self.desc)
+        f.close()
+        f = open(tempdir + "/tags", "w")
+        for tag in self.tags:
+            f.write(tag + "\n")
+        f.close()
+        tar = tarfile.open(tempdir + "/sample.tar.gz", "w:gz")
+        for name in [
+            tempdir + "/data_01.csv",
+            tempdir + "/manifest",
+            tempdir + "/description",
+            tempdir + "/tags",
+        ]:
+            tar.add(name)
+        tar.close()
+        return tempdir + "/sample.tar.gz"
```

Comment:
This needs refactoring to be able to customize with the name. Let's keep this as open as possible, we are working on adding Apache Arrow + Apache Flight which will make dataset streaming E X T R E M E L Y fast (on normal machines, the task will become network bound, not CPU bound as it is currently).

Reply:
100% I hate this class but it gives us the UX we want for now while we 100% replace it with something better later.